### PR TITLE
Initial iOS Support

### PR DESCRIPTION
**Pull request by mikeskydev**
Adds an iOS pipeline focused on Zapbox support, with general fixes for iOS-related functionality.

Co-authored-by: Andy Baker <andy@andybak.net>

The remaining GVR systems have been removed due to incompatibility with iOS. We have switched GVR Audio to use standard unity audio, which we'd need for any modern spatialization system anyways.

### CODE REVIEW

File: .github/workflows/build.yml
Comment (mikeage) [inline .github/workflows/build.yml:306]: This is going to be problematic for people not using CI. At the absolute least, it needs to go into the README!

File: .github/workflows/build.yml
Comment (mikeage) [inline .github/workflows/build.yml:309]: Unless we need it there (perhaps in the packages-lock file) for pinning versions, can we have optional packages not included and add them as needed? (I don't know if this is possible)

File: .github/workflows/build.yml
Comment (mikeage) [inline .github/workflows/build.yml:310]: props for using `tee` for debugging purposes, as opposed to just `> foo.new`

File: Assets/Editor/BuildTiltBrush.cs
Comment (mikeage) [inline Assets/Editor/BuildTiltBrush.cs:65]: Do we need two variables if they both (iOS and Android) get the same value?

File: Assets/Editor/BuildTiltBrush.cs
Comment (mikeage) [inline Assets/Editor/BuildTiltBrush.cs:951]: What happens if we want to install an iOS build from a branch? Does it overwrite the main build?

File: Assets/Scripts/Config.cs
Comment (mikeage) [inline Assets/Scripts/Config.cs:48]: please add a trailing comma so that when we add the next one, we won't need to modify two lines

File: ProjectSettings/ProjectSettings.asset
Comment (mikeage) [inline ProjectSettings/ProjectSettings.asset:11]: Are all of these changes intended?

File: ProjectSettings/ProjectSettings.asset
Comment (mikeage) [inline ProjectSettings/ProjectSettings.asset:141]: deliberate?

File: ProjectSettings/ProjectSettings.asset
Comment (mikeage) [inline ProjectSettings/ProjectSettings.asset:167]: Is this correct?

File: ProjectSettings/ProjectSettings.asset
Comment (mikeage) [inline ProjectSettings/ProjectSettings.asset:867]: Did you mean to remove things from `Standalone`?

File: Assets/ThirdParty/GoogleVR.meta
Comment (mikeage) [inline Assets/ThirdParty/GoogleVR.meta:1]: Did you mean to delete all of this stuff here?

File: .github/workflows/build.yml
Comment (mikeskydev) [reply .github/workflows/build.yml:306]: Good point! this step should be very temporary, I can park this PR until we get a newer SDK version from Zappar.

File: .github/workflows/build.yml
Comment (mikeskydev) [reply .github/workflows/build.yml:309]: See above, but in general unity's manifest.json is quite uncooperative

File: .github/workflows/build.yml
Comment (mikeskydev) [reply .github/workflows/build.yml:310]: I just learned from best! 😉

File: Assets/Editor/BuildTiltBrush.cs
Comment (mikeskydev) [reply Assets/Editor/BuildTiltBrush.cs:65]: I felt this was more appropriate in case of any forks which may have different naming schemes, for example iOS bundles can only have hyphens while android can only have underscores

File: Assets/Editor/BuildTiltBrush.cs
Comment (mikeskydev) [reply Assets/Editor/BuildTiltBrush.cs:951]: I just installed this PR for testing and it installed separately to my local build, so I guess not? anyways, this was causing a build failure related to having brackets in the name

File: ProjectSettings/ProjectSettings.asset
Comment (mikeskydev) [reply ProjectSettings/ProjectSettings.asset:11]: Yes, iOS has a higher refresh rate capability in portrait

File: ProjectSettings/ProjectSettings.asset
Comment (mikeskydev) [reply ProjectSettings/ProjectSettings.asset:141]: build pipeline overwrites this with the data in config.cs, so I felt we should reset it?

File: ProjectSettings/ProjectSettings.asset
Comment (mikeskydev) [reply ProjectSettings/ProjectSettings.asset:167]: Yes

File: ProjectSettings/ProjectSettings.asset
Comment (mikeskydev) [reply ProjectSettings/ProjectSettings.asset:867]: Yes, `UNITY_HAS_GOOGLEVR` is obsolete, especially as we removed the gvr code.

File: Assets/ThirdParty/GoogleVR.meta
Comment (mikeskydev) [reply Assets/ThirdParty/GoogleVR.meta:1]: Yeah, we don't use any of it anymore.

File: Assets/Scripts/Config.cs
Comment (mikeskydev) [reply Assets/Scripts/Config.cs:48]: Will action!

File: .github/workflows/build.yml
Comment (mikeage) [reply .github/workflows/build.yml:306]: I'm ok with pushing it, but I'd recommend a message on Discord as well (pinned).

File: .github/workflows/build.yml
Comment (mikeage) [reply .github/workflows/build.yml:309]: ok, fair enough. I suppose this is also better for the package cache if they're all there.

File: Assets/Editor/BuildTiltBrush.cs
Comment (mikeage) [reply Assets/Editor/BuildTiltBrush.cs:65]: Ok, as long as it's on purpose!

File: Assets/Editor/BuildTiltBrush.cs
Comment (mikeage) [reply Assets/Editor/BuildTiltBrush.cs:951]: Can you add a comment that we can't use brackets (to be clear, I assume you mean what I'd call parentheses -- these things: `()`, right).

This primarily happens when we build from the GUI, where the stamp is `-(menuitem)`. 

Can we just replace those characters instead?

File: Assets/Scripts/VrSdk.cs
Comment (mikeage) [inline Assets/Scripts/VrSdk.cs:43]: let's do a trailing comma here too

File: Assets/Editor/BuildTiltBrush.cs
Comment (mikeskydev) [reply Assets/Editor/BuildTiltBrush.cs:951]: Yep parentheses :D yes I suppose so, I'll change it to just be `-menuitem`

Comment (mikeskydev) [reply Assets/Editor/BuildTiltBrush.cs:951]: Found the actual error: `UnityException: iOS Version has not been set up correctly, it must consist only of '.'s and numbers, must begin and end with a number and be no longer than 18 characters`

**Review by mikeage — APPROVED** [review]
LGTM, unless you want to add the actual fastlane compilation in the same PR

**Comment by mikeskydev** [comment]
We'll leave fastlane for another PR, but was hoping for the zapbox sdk to update so I could resolve the remaining conversation before merging this one

File: .github/workflows/build.yml
Comment (mikeskydev) [reply .github/workflows/build.yml:306]: CC @tangobravo this is our only blocker for merging initial support and getting you some xcode builds :)

**Comment by mikeage** [comment]
I updated this PR based on the changes I recently made. All of the conflicts were resolved in the merge commit

File: .github/workflows/build.yml
Comment (mikeage) [inline .github/workflows/build.yml:237]: Can you rename this to iOS Zapbox, to match the others which is "OS Product" and not "Product OS"?

**Review by mikeage — CHANGES_REQUESTED** [review]

File: .github/workflows/build.yml
Comment (mikeskydev) [reply .github/workflows/build.yml]: done!

File: Assets/Editor/BuildTiltBrush.cs
Comment (mikeage) [inline Assets/Editor/BuildTiltBrush.cs:1796]: great idea ;-)

**Review by mikeage — APPROVED** [review]

**Pull request merged**
